### PR TITLE
Minor refactor: spell out max supported GHC once

### PR DESCRIPTION
**Pull request by ulysses4ever**
This has been requested in the past: https://github.com/haskell/cabal/pull/8666#issue-1531294842

---

**Template B: This PR does not modify behaviour or interface**

*E.g. the PR only touches documentation or tests, does refactorings, etc.*

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Is this a PR that fixes CI? If so, it will need to be backported to older cabal release branches (ask maintainers for directions).

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]

**Comment by geekosaur** [comment]
Hm, how is this going to affect #10656, or vice versa since it's waiting to be merged?

**Comment by ulysses4ever** [comment]
Don't see a problem: I'll just rebase. Effectively this one subsumes https://github.com/haskell/cabal/pull/10656 but I don't want to go into trouble of getting people to agree on closing that one.

File: Cabal/src/Distribution/Simple/GHC.hs [hunk 1]
```diff
@@ -163,14 +163,17 @@ configure verbosity hcPath hcPkgPath conf0 = do
       (userMaybeSpecifyPath "ghc" hcPath conf0)
   let implInfo = ghcVersionImplInfo ghcVersion
 
-  -- Cabal currently supports ghc >= 7.0.1 && < 9.12
+  -- Cabal currently supports GHC up to `maxGhcVersion`
```
Comment (ffaf1) [inline Cabal/src/Distribution/Simple/GHC.hs:166]: Maybe

```suggestion
  -- Cabal currently supports GHC up to `maxGhcVersion` not included.
```

Just to make sure it is clear we are talking about an open interval.

**Review by ffaf1 — APPROVED** [review]

File: Cabal/src/Distribution/Simple/GHC.hs (hunk 1 above)
Comment (geekosaur) [reply Cabal/src/Distribution/Simple/GHC.hs:166]: That should be "inclusive", not "included".

Comment (ulysses4ever) [reply Cabal/src/Distribution/Simple/GHC.hs:166]: "Exclusive" maybe?

Comment (geekosaur) [reply Cabal/src/Distribution/Simple/GHC.hs:166]: I considered that, but it's potentially ambiguous to an ESL speaker as it is sometimes used to mean something roughly opposite to what we intend (think "an exclusive opportunity" and such), whereas the usage here is inspired by its mathematical meaning.

Comment (Bodigrim) [reply Cabal/src/Distribution/Simple/GHC.hs:166]: I'd say "Cabal currently supports GHC < `maxGhcVersion`".

Comment (ulysses4ever) [reply Cabal/src/Distribution/Simple/GHC.hs:166]: I'm not a fan of mixing speech and symbols (there's a lot of advice against it in academic literature). But I hope "less than" is good enough.

**Comment by ulysses4ever** [comment]
@Kleidukos feel free to expedite if desired.

**Review by Mikolaj — APPROVED** [review]

**Comment by ulysses4ever** [comment]
@mergifyio requeue

**Pull request merged**